tests/classes/test_hypergraph.py

```diff
@@ -9,32 +9,36 @@ def test_constructor(edgelist5, dict5, incidence5, dataframe5):
     H_dict = xgi.Hypergraph(dict5)
     H_mat = xgi.Hypergraph(incidence5)
     H_df = xgi.Hypergraph(dataframe5)
+    H_hg = xgi.Hypergraph(H_list)
```

Comment:
Wait, what's the reason to allow create a Hypegraph from a Hypergraph again? 
Something about the subhypergraphs needing to be Hypergraphs?

Reply:
Uuuuuh, I swear I had a good reason when I implemented it, but can't think of it anymore?